src/lib.rs

```diff
@@ -168,7 +202,41 @@ pub async fn run(config: &Config) -> Result<(), ()> {
         Some(s) => s.iter().map(|m| WildMatch::new(m)).collect(),
         None => vec![],
     };
+
+    let gitignored_files: Option<Vec<PathBuf>> = if config.optional.gitignore.is_some() {
+        let files = find_git_ignored_files();
+        println!("Found gitignored files: {:?}", files);
+        files
+    } else {
+        None
+    };
+
+    let is_gitignore_enabled = gitignored_files.is_some();
+
     for link in &links {
+        let canonical_link_source = match fs::canonicalize(&link.source) {
+            Ok(path) => path,
+            Err(e) => {
+                println!("Failed to canonicalize link source: {}. Error: {:?}", link.source, e);
```

Comment:
better use `warn!` macro instead of `println`?

Reply:
Switched to `warn!`